website/docs/d/group.html.markdown

```diff
@@ -0,0 +1,50 @@
+---
+layout: "azuread"
+page_title: "Azure Active Directory: azuread_group"
+sidebar_current: "docs-azuread-datasource-azuread-group"
+description: |-
+  Gets information about an Azure Active Directory group.
+
+---
+
+# Data Source: azuread_group
+
+Gets information about an Azure Active Directory group.
+
+-> **NOTE:** If you're authenticating using a Service Principal then it must have permissions to `Read directory data` within the `Windows Azure Active Directory` API.
+
+## Example Usage (by Object ID)
+
+```hcl
+data "azuread_group" "test_group" {
+  object_id = "78722cfc-8946-11e8-95f1-2200ec79ad01"
+}
+```
+
+## Example Usage (by Group Display Name)
+
+```hcl
+data "azuread_group" "test_group" {
+  name = "MyTestGroup"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `object_id` - (Optional) The ID of the Azure AD Group we want to lookup.
```

Comment:
```suggestion
* `object_id` - (Optional) The UUID of the Azure AD Group
```

Reply:
thinking about this a little - what's the use-case for looking up an AzureAD group by its ID rather than it's name? I can't think the name as the other available output (at this time) would be particularly useful when you've got the ID?

Reply:
The group display names are not unique within AAD. There may be cases where multiple groups have the same name.

Example: 
- User syncs AD forest foo.mydomain.com to Azure AD
- The forest has multiple sub domains, say bar1 and bar2
- Each domain has a domain users group "employees"
- This results in multiple Azure AD groups called "employees" where the only way to target these is the object id.

This is a bad configuration ofc. but it may happen, therefore I thought i might as well give the option to scope by id. Ofc. we don't have to support each (bad) scenario.

So @tombuildsstuff, are we going to keep the option? :-)

Reply:
I was thinking more along the lines of the outputs available, for example if you've only got the group name (and there's a single group matching that name) then the ID is useful as an output; however if you've got the ID I'm not sure of the use-case for what you'd do with the name?

Reply:
Made requested changes in 2ebf0cd